File: myoro_matchup/lib/module/game/domain/dto/game_player_response_dto.dart
import 'package:faker/faker.dart';
import 'package:flutter/foundation.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:myoro_flutter_annotations/myoro_flutter_annotations.dart';

part 'game_player_response_dto.g.dart';

/// Game player response DTO.
@immutable
@myoroModel
@JsonSerializable()
final class GamePlayerResponseDto with _$GamePlayerResponseDtoMixin {
  /// Default constructor.
  const GamePlayerResponseDto({required this.id, required this.username, required this.name});

  /// Fake constructor.
  factory GamePlayerResponseDto.fake() {
    return GamePlayerResponseDto(
      id: faker.randomGenerator.integer(999999),
      username: faker.internet.userName(),
      name: faker.person.name(),
    );
  }

  /// From JSON.
  factory GamePlayerResponseDto.fromJson(Map<String, dynamic> json) {
    return _$GamePlayerResponseDtoFromJson(json);
  }

  /// User ID.
  final int id;

  /// Username.
  final String username;

  /// Name.
  final String name;

  /// To JSON.
  Map<String, dynamic> toJson() {
    return _$GamePlayerResponseDtoToJson(this);
  }
}

File: myoro_matchup_api/src/main/java/com/myoro/myoro_matchup_api/dto/GamePlayerResponseDto.java
package com.myoro.myoro_matchup_api.dto;

/** Game player response DTO. */
public class GamePlayerResponseDto {
  /** User ID. */
  private Long id;

  /** Username. */
  private String username;

  /** Name. */
  private String name;

  /** Default constructor. */
  public GamePlayerResponseDto() {}

  /** Constructor with all fields. */
  public GamePlayerResponseDto(Long id, String username, String name) {
    this.id = id;
    this.username = username;
    this.name = name;
  }

  /**
   * Getter for id
   *
   * @return the user ID
   */
  public Long getId() {
    return id;
  }

  /**
   * Getter for username
   *
   * @return the username
   */
  public String getUsername() {
    return username;
  }

  /**
   * Getter for name
   *
   * @return the name
   */
  public String getName() {
    return name;
  }

  /** Setter for id. */
  public void setId(Long id) {
    this.id = id;
  }

  /** Setter for username. */
  public void setUsername(String username) {
    this.username = username;
  }

  /** Setter for name. */
  public void setName(String name) {
    this.name = name;
  }
}
